runtime/jcl/common/sun_reflect_ConstantPool.c

```diff
@@ -267,11 +270,21 @@ getMethodAt(JNIEnv *env, jobject constantPoolOop, jint cpIndex, UDATA resolveFla
 				methodID = (jmethodID) vmFunctions->getJNIMethodID(vmThread, method);
 			}
 		}
-		vmFunctions->internalReleaseVMAccess(vmThread);
-	}
 
-	if (NULL != methodID) {
-		returnValue = (*env)->ToReflectedMethod(env, constantPoolOop, methodID, J9CPTYPE_STATIC_METHOD == cpType);
+		if (NULL != methodID) {
+			j9object_t classObject = J9CLASS_FROMJ9CLASSREF(vmThread, constantPoolOop)->classObject;
+			jclass jlClass = vmFunctions->j9jni_createLocalRef(env, classObject);
+			if (NULL != jlClass) {
+				vmFunctions->internalReleaseVMAccess(vmThread);
+				returnValue = (*env)->ToReflectedMethod(env, jlClass, methodID, J9CPTYPE_STATIC_METHOD == cpType);
```

Comment:
Is this going to be a problem since you've mapped back to the jlClass which may have the wrong J9Class?